src/lib/OpenEXRCore/internal_util.h

```diff
@@ -42,4 +42,19 @@ compute_sampled_lines (int height, int y_sampling, int start_y)
     return nlines;
 }
 
+static inline int
+compute_sampled_width (int width, int x_sampling, int start_x)
+{
+    /*
+     * we require that the start_x % x_sampling == 0 and for tiled images (and for deep),
```

Comment:
oh jeepers, this requirement was driving me crazy because I couldn't tell if it was actually a requirement or not. 

do we also require that tiled images have a width that's a multiple of the tile size? I'd like to imagine the answer is yes, because writing the code for not-a-multiple was omg a mighty pain

Reply:
a mighty pain because instead of just reading the dang tiles you have to stage a fake tile in case there's only a few pixels in the last one. it makes everything twice as much code as it wants to be.

Reply:
The width is not required to match a multiple of the tile size, but the chunk info will tell you when that is the case, but all tiles from data window min coordinate and proceed from there... I have encountered people who added "overscan" to the tile set in production :( was an accident mostly, but the system was happily chunking away

Reply:
> a mighty pain because instead of just reading the dang tiles you have to stage a fake tile in case there's only a few pixels in the last one. it makes everything twice as much code as it wants to be.

not true! if your base tile is 32x32, but there's only 10x2 left or whatever at the bottom right of the tile, you only have to give that width / height of memory to the core and it won't overwrite memory on you :)

Reply:
I'll revisit my tile reader!